monai/networks/blocks/attention.py

```diff
@@ -0,0 +1,327 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import Optional, Tuple
+
+import torch
+import torch.nn as nn
+
+from monai.networks.layers.utils import get_rel_pos_embedding_layer
+from monai.utils import optional_import
+
+Rearrange, _ = optional_import("einops.layers.torch", name="Rearrange")
+
+
+class SABlock(nn.Module):
```

Comment:
Does this move break anything? If somewhere else calls the SABlock in selfattention directly?

Reply:
no its just the transformerblock that imports it directly. the transformer block is then used by a few networks (ViT, SwinUNET)

Reply:
May not break anything in monai but may affect users?
Perhaps we need add `from monai.networks.blocks.attention import SABlock` in selfattention and add a deperecated warning? What do you think?